plugins/lasersensor/src/LaserSensor.cc

```diff
@@ -127,8 +146,11 @@ void GazeboYarpLaserSensor::Load(sensors::SensorPtr _sensor, sdf::ElementPtr _sd
     }
     else if(!driver_properties.check("yarpDeviceName"))
     {
+        #ifndef GAZEBO_YARP_PLUGINS_DISABLE_IMPLICIT_NETWORK_WRAPPERS
         scopedDeviceName = sensorName + "::" + driver_list[0]->key;
+        #endif
         yCError(GAZEBOLASER)<<"failed getting yarpDeviceName parameter value";
+        return;
```

Comment:
Unless I am missing something, this should only be here if `GAZEBO_YARP_PLUGINS_DISABLE_IMPLICIT_NETWORK_WRAPPERS` is defined.

Reply:
Should be this `ifndef` of `ifdef` ?

Reply:
ifndef since I added it later, before was without return

Reply:
Sorry, the `return` should be there only if the `GAZEBO_YARP_PLUGINS_DISABLE_IMPLICIT_NETWORK_WRAPPERS` is `ON`, so I guess that should be an `ifdef`?

Reply:
We can discuss of this on Teams if you prefer.

Reply:
Ok, sorry I was confused. Yes it should be ifdef

Reply:
I think now it is clearer